syntax = 'proto3';

package ModelOrchestrator;
option go_package = '/pkg/internal/proto';

import "google/protobuf/timestamp.proto";
import "common.proto";

service ModelOrchestrator {
  rpc SendData(stream ProcessedData) returns (common.Status) {}
}

message ProcessedData {
  repeated int32 source_id = 1;
  int32 model_id = 2;
  string data_json = 3;
  google.protobuf.Timestamp arrival_time = 4;
}
